### PR TITLE
Horizon engines now actually shoot fire

### PR DESCRIPTION
**Pull request by ShiftyRail**
engines at full speed

Also reshuffled a bit the pipe networks so that Atmosians have more play space now that the engine zone is hazardous.

image

The  engines fire a straight  line of fire every 30 seconds, so someone fast enough can very well cross them between cycles.

~~I plan to maybe do more stuff with Horizon engines and their subsystem eventually.~~ 

:cl:
- tweak: NRV Horizon will now actually consume fuel when travelling.

### CODE REVIEW

**Comment by SageOfLight** [comment]
>the backside fires

uhhhhhhhhhhhhhhhhhhhhh should probably put some tiles or obstruction there there so it doesn't look assy

**Comment by ShiftyRail** [comment]
The escape shuttle also has this kind of backside fire where the heaters get doused in flame too.

**Comment by west3436** [comment]
Do the engines ever turn off? If not why not remove the catwalks that are in the path of the flame? Just keeping it for the autism crabs?

**Comment by ShiftyRail** [comment]
I'm going to try and rework the area so that it looks less artificial. The engines shoot in cycles of 30 seconds, so a fast human can use it as a shortcut or even throw off a pursuit with the wall of fire. 

The flames always shoot straight, so you can hide in-between engines too. Nobody will ever do that, but (I think) it's cool.

**Comment by BomberBro** [comment]
If a dpush wizard pushed an engine part or a heater does this mean it'll be constantly streaming out instakill flame in the halls?

**Comment by ShiftyRail** [comment]
He would indeed need to dpush the entire engine part. There can be a fix to that.

**Comment by ShiftyRail** [comment]
image

Changed the aesthetic of the ass of the ship a little bit, made it so engines do not roar when dimensionally pushed, and tried to remove the backblast.

**Comment by west3436** [comment]
>ass of the ship

**Comment by ShiftyRail** [comment]
It's a nautical term.

**Comment by dunsel-hordon** [comment]
will the fire affect cooling loops? As in if i have some cooling pipes there will it make the gasses in them heat up?

**Comment by ShiftyRail** [comment]
The pipes to cool down the shard have been pushed sideways, and I will add a pre-made connection between the two sides of the ship. The fire does not heat up space, but  it's not recommended to work around the engines.

**Comment by ShiftyRail** [comment]
@dunsel-hordon 

horizon

Here is how the atmos cooling area looks like.

**Comment by ShiftyRail** [comment]
Mapping conflicts fixed.

**Comment by Zth--** [comment]
This one has map priority but it's still conflicting. I'm thinking about removing the tag and merging #27360 

Okay?

**Comment by ShiftyRail** [comment]
Why yes, solving map conflicts and having that work erased IS after all the greatest coding experience a man can have

**Comment by Zth--** [comment]
but is it really deleted

**Comment by Kurfursten** [comment]
You do have map priority now so please take some action

The data isn't actually lost, you can back up your current file on your desktop and copy/paste it over to the current map in conflict resolution

**Comment by Kurfursten** [comment]
It seems you are busy right now and unable to work on conflicts. That's okay, we'll loop back around to you. I'm giving you advance notice of the priority order so you don't accidentally fix them too early and get slapped with conflicts a second time.

New mapping priority
#27358
#27360
#27356 
#27287

**Comment by Kurfursten** [comment]
Your PR is now on deck for mapping priority.  You will receive mapping priority after #27356

**Comment by Kurfursten** [comment]
You have map priority. Please let me know if you don't intend to use it so I can pass it on to #27492

I'm going to check back on this one in 48 hours

**Comment by ShiftyRail** [comment]
Thanks, but I pass my turn. Cannot work on this until Saturday 15 at
earliest.

File: code/game/shuttle_engines.dm [hunk 1]
```diff
@@ -129,12 +130,14 @@
...
-/obj/structure/shuttle/engine/propulsion/proc/shoot_exhaust(forward=9, backward=9)
+/obj/structure/shuttle/engine/propulsion/proc/shoot_exhaust(forward=9, backward=9, var/turf/source_turf)
 	if(!anchored)
 		return
 	var/turf/target = get_edge_target_turf(src,dir)
-	var/turf/T = get_turf(src)
-	var/obj/item/projectile/fire_breath/A = new /obj/item/projectile/fire_breath/shuttle_exhaust(T)
+	var/turf/T = source_turf
+	if (!source_turf)
+		source_turf = get_turf(src)
```
Comment (Exxion) [inline code/game/shuttle_engines.dm:139]: what

**Pull request merged**